src/python/WMCore/MicroService/MSPileup/MSPileupData.py

```diff
@@ -165,6 +191,16 @@ def getPileup(self, spec, projection=None):
         """
         results = []
 
+        # remove API key from the spec as it defines pileup API
+        if isinstance(spec, dict) and 'API' in spec:
+            spec.pop('API')
+
+        self.logger.info("getPileup spec=%s, projection=%s", spec, projection)
```

Comment:
I'd suggest to move this line below the next if statement. Maybe rephrase as well to something like "Getting pileup with spec: %s, projection: %s"

Reply:
done